src/components/filePreview/filePreview.tsx

```diff
@@ -17,17 +22,54 @@ export default function FilePreview(
 ) {
   const theme = useTheme()
   const maximumFileNameLength = 15
+
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  function handleFileClick() {
+    setIsModalOpen(true)
+  }
+
+  function handleCloseModal() {
+    setIsModalOpen(false)
+  }
+
+  const isPdfFile =
+    props.file.url && props.file.url.toLowerCase().endsWith('.pdf')
+
   return (
-    <Card
-      className="rustic-file-preview"
-      data-cy="file-preview"
-      variant="outlined"
-      sx={{ boxShadow: theme.shadows[1] }}
-    >
-      <Typography variant="subtitle2" data-cy="file-name">
-        {shortenString(props.file.name, maximumFileNameLength)}
-      </Typography>
-      {props.children}
-    </Card>
+    <>
+      <Card
+        className={`rustic-file-preview${isPdfFile ? ' rustic-cursor-pointer' : ''}`}
+        data-cy="file-preview"
+        variant="outlined"
+        sx={{ boxShadow: theme.shadows[1] }}
+        onClick={handleFileClick}
+      >
+        <Typography variant="subtitle2" data-cy="file-name">
+          {shortenString(props.file.name, maximumFileNameLength)}
+        </Typography>
+        <div onClick={(e) => e.stopPropagation()}>{props.children}</div>
+      </Card>
+      {isPdfFile && (
+        <Modal
+          open={isModalOpen}
+          onClose={handleCloseModal}
+          className="rustic-pdf-viewer-modal"
```

Comment:
Maybe I'm overthinking but in the future I think we might be able to have other files that can be previewed (docs for example). If that's the case this modal container could also be used for those, so I was thinking this classname could be more generalized. Same with the child div. What do you think?